apps/web/src/routes/dashboard/_dashboard/-components/booking-table.tsx

```diff
@@ -0,0 +1,185 @@
+
+  const blob = new Blob([csvData], { type: 'text/csv' });
+  const link = document.createElement('a');
+  link.href = URL.createObjectURL(blob);
+
+  const sanitizedEventName = eventName
+    .replaceAll(/[^\da-z]/gi, '_')
+    .toLowerCase();
+  const filename = `${sanitizedEventName}_participants.csv`;
+  link.download = filename;
+  link.click();
+};
+
+const BookingTable = () => {
+  const { t } = useTranslation();
+
+  const { data: allEvents, isFetched } =
+    trpc.user.calendar.getCalendarEvents.useQuery({
+      language: 'en',
+      upcomingEvents: true,
+    });
+
+  return (
+    <div>
+      {!isFetched && <Loader size={'s'} />}
+
+      <div className="overflow-x-auto max-md:hidden md:flex">
+        <Table className="!max-w-[910px]">
+          <TableHeader className="border-b-0">
+            <TableRow>
+              <TableHead className="w-[137px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.type')}
+              </TableHead>
+              <TableHead className="w-[420px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.nameOfEvent')}
+              </TableHead>
+              <TableHead className="w-[180px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.date')}
+              </TableHead>
+              <TableHead className="flex justify-center w-[180px] text-base font-medium text-dashboardSectionTitle">
+                {t('dashboard.adminPanel.listOfParticipants')}
+              </TableHead>
+            </TableRow>
+          </TableHeader>
+          <TableBody>
+            {allEvents?.map((event) => (
+              <EventRow key={event.id} event={event} />
+            ))}
+          </TableBody>
+        </Table>
+      </div>
+
+      <div className="flex flex-col gap-4 md:hidden">
+        {allEvents?.map((event) => <EventCard key={event.id} event={event} />)}
+      </div>
+    </div>
+  );
+};
+
+interface EventProps {
+  event: Event;
+}
+
+const EventRow = ({ event }: EventProps) => {
+  const { t } = useTranslation();
+
+  const { data: participants, isFetched } =
+    trpc.user.events.getParticipantsForEvent.useQuery({
+      eventId: event.id,
+    });
+
+  const startDate = event.startDate || new Date();
+
+  return (
+    <TableRow>
+      {!isFetched && <Loader size={'s'} />}
+      <TableCell>{event.type}</TableCell>
+      <TableCell>{event.name}</TableCell>
+      <TableCell>{formatDateWithoutTime(startDate)}</TableCell>
+      <TableCell>
+        <Button
+          size="s"
+          className="mx-auto"
+          onClick={() => handleDownload(participants ?? [], event.name)}
+        >
+          {t('dashboard.adminPanel.downloadList')}
+        </Button>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+const EventCard = ({ event }: EventProps) => {
+  const { t } = useTranslation();
+
+  const { data: participants, isFetched } =
```

Comment:
Same here, by doing request for each event individually we have to duplicate code and create many small requests